docs/tutorials/services/source-ip.md

```diff
@@ -0,0 +1,338 @@
+
+```console
+$ kubectl expose deployment source-ip-app --name=clusterip --port=80 --target-port=8080
+service "clusterip" exposed
+
+$ kubectl get svc clusterip
+NAME         CLUSTER-IP    EXTERNAL-IP   PORT(S)   AGE
+clusterip    10.0.170.92   <none>        80/TCP    51s
+```
+
+And hitting the `ClusterIP` from a pod in the same cluster
+
+```console
+$ kubectl run busybox -it --image=busybox --restart=Never --rm
+Waiting for pod default/busybox to be running, status is Pending, pod ready: false
+If you don't see a command prompt, try pressing enter.
+
+# ip addr
+1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue
+    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
+    inet 127.0.0.1/8 scope host lo
+       valid_lft forever preferred_lft forever
+    inet6 ::1/128 scope host
+       valid_lft forever preferred_lft forever
+3: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1460 qdisc noqueue
+    link/ether 0a:58:0a:f4:03:08 brd ff:ff:ff:ff:ff:ff
+    inet 10.244.3.8/24 scope global eth0
+       valid_lft forever preferred_lft forever
+    inet6 fe80::188a:84ff:feb0:26a5/64 scope link
+       valid_lft forever preferred_lft forever
+
+# wget -qO - 10.0.170.92
+CLIENT VALUES:
+client_address=10.244.3.8
+command=GET
+...
+```
+
+### Source IP for Services with Type=NodePort
+
+As of Kubernetes 1.5, packets sent to Services with [Type=NodePort](/docs/user-guide/services/#type-nodeport)
+are source NAT'd by default. You can test this by creating a `NodePort`
+Service
+
+```console
+$ kubectl expose deployment source-ip-app --name=nodeport --port=80 --target-port=8080 --type=NodePort
+service "nodeport" exposed
+
+$ NODEPORT=$(kubectl get -o jsonpath="{.spec.ports[0].nodePort}" services nodeport)
+$ NODES=$(kubectl get nodes -o jsonpath='{ $.items[*].status.addresses[?(@.type=="ExternalIP")].address }')
+```
+
+if you're running on a cloudprovider, you need to open up a firewall-rule for
+the `nodes:nodeport` reported above.
+Now you can try reaching the Service from outside the cluster through the node
+port allocated above.
+
+```console
+$ for node in $NODES; do curl -s $node:$NODEPORT | grep -i client_address; done
+client_address=10.180.1.1
+client_address=10.240.0.5
+client_address=10.240.0.3
+```
+
+Note that these are not your IPs, they're cluster internal IPs. This is what happens:
+
+* Client sends packet to `node2:nodePort`
+* `node2` replaces the source IP on the packet with its own (SNAT)
+* `node2` replaces the destination IP on the packet with the pod IP
+* the pod IP is routed to node 1, and then to the endpoint
+* the pod's reply is routed back to node2
+* the pod's reply is sent back to the client
+
+Visually:
+
+```
+          client
+             \ ^
+              \ \
+               v \
+   node 1 <--- node 2
+    | ^   SNAT
+    | |   --->
+    v |
+ endpoint
+```
+
+
+To avoid this, we have a beta feature in 1.5, triggered by the `service.beta.kubernetes.io/external-traffic`
```

Comment:
Remove the "We" construct and the prounoun. "To avoid this" is ambiguous; say explicitly what the feature is designed to avoid.

"To avoid (doing whatever it is), Kubernetes version 1.5 introduced the (whatever the annotation is). Setting this annotation to the value `OnlyLocal`..."

After the paragraph, include a blockquote note like this one:

> **Note:** The (annotation) is a beta feature in Kubernetes 1.5.